### ISSUE TITLE
kasada is not working with 

### ISSUE THREAD

**Issue by orithecapper** [ulixee/unblocked#68]
![image](https://github.com/ulixee/unblocked/assets/131520229/78c20667-ab86-4124-a7f8-26e133cf9e7b)

**Comment by orithecapper** [comment]
```js


const { Agent } = require('@ulixee/unblocked-agent');

(async () => {
  const agent = new Agent({
    // browserEngine: new Chrome113(),
    options: {
      showChrome: true,
    },
  });
  agent.hook({
    onNewBrowser(browser) {
      browser.engine.launchArguments.push('--no-startup-window', '--disable-background-networking');
    },
  });
  const page = await agent.newPage();
  await page.goto('https://www.canadagoose.com/us/en/trillium-parka-heritage-6660L.html');
  await page.waitForLoad('PaintingStable');

  // const outerHTML = await page.mainFrame.outerHTML();
  // const title = await page.evaluate('document.title');
  // const intro = await page.evaluate(`document.querySelector('p').textContent`);
  // console.log(title)
  // await agent.close();
})();
```

**Comment by blakebyrnes** [comment]
Your setup seems to not include any of the plugins that make the browser look like a consumer browser. You might find it easier to use Hero, which automatically included both the default browser emulator and the human emulator.

**Comment by pashkasan** [comment]
i told u i will come back)
lack of examples how to use extensive way, like using agent with default browser emulation and different browsers
example provided is nothing, hard to implement,not all super coders) show how to use power of unlocked, that the way
and expose cdp and how is cast postmessage events??

**Comment by blakebyrnes** [comment]
@pashkasan yeah, this library is not well documented because it's kind of an internal system for Hero (https://github.com/ulixee/hero) (https://ulixee.org/docs/hero). It's meant to be a super-coder library if you want to get low level. Ideally, would love to document it so it can be usable too, but it's hard to use without the niceties of Hero, so it's not currently a high priority.

**Comment by pashkasan** [comment]
i can send u private codes of puppetter, how to catch not popup but css popups many sites use it now and lock screen

**Comment by pashkasan** [comment]
for sure u need more usabale example how to use power of your project to emulate real user, ua and etc this will grow community and codefor sure  more coders MORE POWER

**Comment by blakebyrnes** [comment]
> for sure u need more usabale example how to use power of your project to emulate real user, ua and etc this will grow community and codefor sure more coders MORE POWER

Ah, yes. Would be very helpful. Are you using low level Agent? Not Hero? There's a little more documentation on Hero (although we could certainly use more)

**Comment by pashkasan** [comment]
Flow handlers, i  read this article, but not find solution to my question
how to handle event in nodejs
ie window.addEventListener('message', e => {console.log('ev', e);});

**Comment by pashkasan** [comment]
> > for sure u need more usabale example how to use power of your project to emulate real user, ua and etc this will grow community and codefor sure more coders MORE POWER
> 
> Ah, yes. Would be very helpful. Are you using low level Agent? Not Hero? There's a little more documentation on Hero (although we could certainly use more)

yes, low level for now, tried you project several times, no luck, but looking to use it with agent, not hero i think it overloaded with command center

**Comment by pashkasan** [comment]
what i need = low cpu profile, raw cdp for some moments, browser emulator u implemented with great mitm

**Comment by blakebyrnes** [comment]
> Flow handlers, i read this article, but not find solution to my question
> how to handle event in nodejs
> ie window.addEventListener('message', e => {console.log('ev', e);});

Flow Handlers were an answer to your comment about css popups. If you want to run raw javascript on the page, you would just use `page.evaluate("window.addEventListener('message', e => {console.log('ev', e);});")`

**Comment by blakebyrnes** [comment]
> what i need = low cpu profile, raw cdp for some moments, browser emulator u implemented with great mitm

If you'd like help, maybe the best option is to connect on Discord. I understand your desire to have better documentation on Agent. I'm not sure Agent is going to have a smooth user experience in the near-term though, as almost all of our dom handling is on the Hero side.

**Comment by pashkasan** [comment]
yes, but need to catch in nodejs soft
pptr

        const listenForPageEvent = function (page, type) {
            log('listenForPageEvent init for ', page.url());
            return page.evaluate(type => {
                window.addEventListener(type, e => {

                    window.onMessageReceivedEvent({
                        type,
                        event: e.data,
                        origin: e.origin
                    });
                });
            }, type);
        };

        await page.exposeFunction('onMessageReceivedEvent', e => {
            
            let event = e.event;
            
            switch (event) {

		got it

**Comment by pashkasan** [comment]
i will do

**Comment by orithecapper** [comment]
> Your setup seems to not include any of the plugins that make the browser look like a consumer browser. You might find it easier to use Hero, which automatically included both the default browser emulator and the human emulator.

Hey, both of them didnt work... 

for ref, I tried using my computer@work , and it did work 

but when for example trying to go to nike.com and "add cart" they do some how detect that its a virtualised thing

kasada is really impresive I think they have some interesting way to detect fingerprints

**Comment by orithecapper** [comment]
> Your setup seems to not include any of the plugins that make the browser look like a consumer browser. You might find it easier to use Hero, which automatically included both the default browser emulator and the human emulator.

Hero fails to emulate , 
also nike.com "Add cart" is failing hard too not that I care much, its just a point that ulixee is being detected somehow 
imo you can make a lot of money if you can figure out what they do , and sell your own bot protection and not have many attacks know...


anyways, about kasada, they say that they want to make "Attacking too costly" and "waste attacker resources" 
in their blogs.. you may be interested
